lib/supavisor/secret_checker.ex

```diff
@@ -0,0 +1,117 @@
+defmodule Supavisor.SecretChecker do
+  @moduledoc false
+
+  use GenServer
+  require Logger
+
+  alias Supavisor.Helpers
+
+  @interval :timer.seconds(15)
+
+  def start_link(args) do
+    name = {:via, Registry, {Supavisor.Registry.Tenants, {:secret_checker, args.id}}}
+
+    GenServer.start_link(__MODULE__, args, name: name)
+  end
+
+  def init(args) do
+    Logger.debug("SecretChecker: Starting secret checker")
+    tenant = Supavisor.tenant(args.id)
+
+    [%{auth: auth, user: user} | _] =
+      Enum.filter(args.replicas, fn e -> e.replica_type == :write end)
```

Comment:
`Enum.find/2` wouldn't be better there? In the end we are interested only in first entry.

Reply:
to stop looping when the first e.replica_type == :write is found?